components/webui/imports/api/search/server/methods.js

```diff
@@ -158,17 +190,26 @@ Meteor.methods({
     /**
      * Cancels an ongoing search operation identified by jobId.
      *
-     * @param {number} jobId of the search operation to cancel
+     * @param {number} searchJobId
+     * @param {number} aggregationJobId
      */
-    async "search.cancelOperation"({
-        jobId,
+    async "search.cancelOperation" ({
+        searchJobId,
+        aggregationJobId,
     }) {
-        logger.info("search.cancelOperation jobId =", jobId);
+        logger.info(`search.cancelOperation searchJobId=${searchJobId}, ` +
+            `aggregationJobId=${aggregationJobId}`);
 
         try {
-            await searchJobsDbManager.submitQueryCancellation(jobId);
+            await searchJobsDbManager.submitQueryCancellation(searchJobId);
+            await searchJobsDbManager.submitQueryCancellation(aggregationJobId);
+
+            await searchJobCollectionsManager.dropCollection(searchJobId);
+            await searchJobCollectionsManager.dropCollection(aggregationJobId);
```

Comment:
Hm, this might cause issues with the backend still writing results after the collections have been dropped. Do we really need to call it here?

Reply:
I agree. It should only be done in `search.clearResults`, or the existing results are not even viewable when the user manually cancel the query. Sorry for the oversight.